src/libtools/graphalgorithm.cc

```diff
@@ -248,52 +237,43 @@ ReducedGraph reduceGraph(Graph graph) {
 
 vector<Graph> decoupleIsolatedSubGraphs(Graph graph) {
 
-  list<int> vertices_list = vectorToList_(graph.getVertices());
-  vector<Graph> subGraphs;
-  Graph_BF_Visitor graph_visitor_breadth_first;
-  graph_visitor_breadth_first.setStartingVertex(vertices_list.front());
-  if (singleNetwork(graph, graph_visitor_breadth_first)) {
-    subGraphs.push_back(graph);
-    return subGraphs;
-  }
+ const std::vector<int>& vertices = graph.getVertices();
+ //bool vector to see if vertex is already part of graph
+ std::vector<bool> vertex_analysed=std::vector<bool>(vertices.size(),false);
 
-  list<int>::iterator vertices_iterator = vertices_list.begin();
-  while (vertices_iterator != vertices_list.end()) {
-    unordered_map<int, GraphNode> sub_graph_nodes;
+ std::vector<Graph> subGraphs;
+ unsigned i=0;
+ while(i<vertices.size()){
+    if(vertex_analysed[i]){i++;continue;}
```

Comment:
This if statement should probably be a while, or something of the sort. You do not want to explore any of the vertices if they have already been explored. Stop when when you hit a vertex that has not been explored or there are none left to explore.

Reply:
while(vertex_analysed[i] && i<vertices.size()){ i++;continue}

Or something like it, then you will have to check that you have not reached the end of the available vertices before you continue. 

Nevermind, I didn't realize what the continue keyword did.

Reply:
Ya, I think in the future I will avoid deleting elements in a loop while iterating over them.